scheduler/scheduler.go

```diff
@@ -198,9 +201,16 @@ func (s *Scheduler) prepareAndExec(r gaia.PipelineRun) {
 	// Create new plugin instance
 	pS := s.pluginSystem.NewPlugin(s.ca)
 
-	// Connect to plugin(pipeline)
+	// Init the plugin
 	path = filepath.Join(path, gaia.LogsFileName)
-	if err := pS.Connect(c, &path); err != nil {
+	if err := pS.Init(c, &path); err != nil {
+		gaia.Cfg.Logger.Debug("cannot initialize the plugin", "error", err.Error(), "pipeline", pipeline)
+		s.finishPipelineRun(&r, gaia.RunFailed)
+		return
+	}
+
+	// Connect to plugin(pipeline)
+	if err := pS.Validate(); err != nil {
```

Comment:
The error message that follows this, is no longer valid. It didn't fail because it couldn't connect. It failed because the validation failed whatever that is in the plugin?